Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/WalletManager.java

```diff
@@ -763,158 +765,133 @@ private void handleTransferDeleted(CoreBRCryptoWallet coreWallet, CoreBRCryptoTr
+                        Log.e(TAG, "BRCryptoCWMBtcGetTransactionsCallback received an error, completing with failure", error);
+                        manager.announceTransactionComplete(handle, false);
+                    }
+                });
     }
 
-    private void btcSubmitTransaction(Pointer context, BRWalletManager managerImpl, Pointer walletImpl,
-                                      BRTransaction transactionImpl, int rid) {
-        Log.d(TAG, "BRSubmitTransactionCallback");
+    private void btcSubmitTransaction(Pointer context, BRCryptoWalletManager manager, BRCryptoCallbackHandle handle,
+                                      Pointer tx, SizeT txLength) {
+        Log.d(TAG, "BRCryptoCWMBtcSubmitTransactionCallback");
+
+        int transactionLength = UnsignedInts.checkedCast(txLength.longValue());
+        byte[] transaction = tx.getByteArray(0, transactionLength);
 
-        CoreBRTransaction transaction = CoreBRTransaction.create(transactionImpl);
-        query.putTransaction(getNetwork().getUids(), transaction.serialize(), new CompletionHandler<Transaction>() {
+        query.putTransaction(getNetwork().getUids(), transaction, new CompletionHandler<Transaction>() {
             @Override
             public void handleData(Transaction data) {
-                Log.d(TAG, "BRSubmitTransactionCallback: succeeded");
-                managerImpl.announceSubmit(rid, transaction, 0);
+                Log.d(TAG, "BRCryptoCWMBtcSubmitTransactionCallback: succeeded");
+                manager.announceSubmit(handle, true);
             }
 
             @Override
             public void handleError(QueryError error) {
-                Log.e(TAG, "BRSubmitTransactionCallback: failed", error);
-                managerImpl.announceSubmit(rid, transaction, 1);
+                Log.e(TAG, "BRCryptoCWMBtcSubmitTransactionCallback: failed", error);
+                manager.announceSubmit(handle, false);
             }
         });
     }
 
-    private List<String> getAllAddrs(BRWalletManager manager) {
-        final UnsignedInteger UNUSED_ADDR_LIMIT = UnsignedInteger.valueOf(25);
-
-        manager.generateUnusedAddrs(UNUSED_ADDR_LIMIT);
-        List<String> addrs = new ArrayList<>(manager.getAllAddrs());
-        addrs.addAll(manager.getAllAddrsLegacy());
-        return addrs;
-    }
-
     // ETH client
 
-    private void ethGetBalance(Pointer context, BREthereumEwm ewm, BREthereumWallet wid, String address, int rid) {
-        Log.d(TAG, "BREthereumClientHandlerGetBalance");
+    private void ethGetEtherBalance(Pointer context, BRCryptoWalletManager manager, BRCryptoCallbackHandle handle, String networkName, String address) {
+        Log.d(TAG, "BRCryptoCWMEthGetEtherBalanceCallback");
 
-        BREthereumNetwork network = ewm.getNetwork();
-        String networkName = network.getName().toLowerCase();
+        // TODO(discuss): Do we actually need a rid here? Can it just be random? Does it need to tie into our C rid?
```

Comment:
Big question here; does the RID matter? Does it need to be the same RID as what the Core has?